test/integration/draining_test.exs

```diff
@@ -6,18 +6,47 @@ defmodule Oban.Integration.DrainingTest do
   @oban_opts repo: Repo, queues: false
 
   test "all jobs in a queue can be drained and executed synchronously" do
-    start_supervised!({Oban, @oban_opts})
+    with_supervised_oban(fn ->
```

Comment:
I like this pattern, but I'd rather keep it out of this PR and do it all in one go. Most of the integration tests would benefit from a `with_supervised_oban` helper defined in the `Oban.Case` module.